### PR TITLE
Fix whisper doc

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Fixes whisper doc-test. I used `add_code_sample_docstrings` but didn't properly check that it does not support whisper model (which has to be given `input_ids`)

### CODE REVIEW

File: src/transformers/models/whisper/modeling_whisper.py [hunk 1]
```diff
@@ -1028,7 +1012,23 @@ def forward(
         output_hidden_states=None,
         return_dict=None,
     ):
-
+        r"""
+        Returns: Example:
+         ```python
+         >>> import torch
+         >>> from transformers import WhisperModel, WhisperFeatureExtractor
```
Comment (ydshieh) [inline src/transformers/models/whisper/modeling_whisper.py:1019]: Let's put `WhisperFeatureExtractor` before `WhisperModel`

**Review by ydshieh — APPROVED** [review]
LGTM, thanks @ArthurZucker . Make sure you run the doctest, so we are 100% get rid of this test failure :-) 🙏

File: src/transformers/models/whisper/modeling_whisper.py [hunk 2]
```diff
@@ -1028,7 +1012,23 @@ def forward(
         output_hidden_states=None,
         return_dict=None,
     ):
-
+        r"""
+        Returns: Example:
```
Comment (sgugger) [inline src/transformers/models/whisper/modeling_whisper.py:1016]:
```suggestion
        Returns:
        
        Example:
```
The returns statement needs to be on its own line.

File: src/transformers/models/whisper/modeling_whisper.py [hunk 3]
```diff
@@ -1004,14 +996,6 @@ def freeze_encoder(self):
         self.encoder._freeze_parameters()
 
     @add_start_docstrings_to_model_forward(WHISPER_INPUTS_DOCSTRING)
-    @add_code_sample_docstrings(
```
Comment (sgugger) [inline src/transformers/models/whisper/modeling_whisper.py:1007]: This needs to be changed to use `replace_return_xxx` (forgot the exact name) instead.

**Review by sgugger — COMMENTED** [review]
You can't jsut fully remove those decorators @ArthurZucker, you then need to use the other one that replaces the word "Returns" by the whole documentation of the specific model output.

**Comment by ArthurZucker** [comment]
Thanks, I missed that 😅

**Comment by ydshieh** [comment]
> Thanks, I missed that 😅

Me too, I am bad guy!

File: src/transformers/models/whisper/modeling_whisper.py [hunk 4]
```diff
@@ -1004,14 +996,7 @@ def freeze_encoder(self):
...
-        processor_class=_PROCESSOR_FOR_DOC,
-        checkpoint=_CHECKPOINT_FOR_DOC,
-        output_type=Seq2SeqModelOutput,
-        config_class=_CONFIG_FOR_DOC,
-        expected_output=_EXPECTED_OUTPUT_SHAPE,
-        modality="audio",
-    )
+    @replace_return_docstrings(output_type=Seq2SeqModelOutput)
```
Comment (ydshieh) [inline src/transformers/models/whisper/modeling_whisper.py:999]: also need `, config_class=_CONFIG_FOR_DOC)`, just like in other modeling files :-) I am a good guy now

Comment (ArthurZucker) [reply src/transformers/models/whisper/modeling_whisper.py:999]: Seems to work without it, but will add it eitherway

Comment (ydshieh) [reply src/transformers/models/whisper/modeling_whisper.py]: it's probably will miss in the rendered doc page (if we don't include). Never try myself, can't say 100% sure.

**Pull request merged**